### PR TITLE
IHDP-19 Upgrade Infinispan to 9.4.1.Final

### PR DESCRIPTION
**Pull request by pruivo**
https://issues.jboss.org/browse/IHDP-19

Initial backport of downstream changes.

* I didn't see any reason to have infinispan as a parent. I removed that.
* Javadoc isn't generated automatically. Should it be?
* I've aligned the dependencies with infinispan 9.4.1.Final
* Should we create a `distribution` profile to create the zip and javadoc?

### CODE REVIEW

File: core/pom.xml [hunk 1]
```diff
@@ -16,6 +16,7 @@
     <description>InputFormat/OutputFormat to integrate with Hadoop Map Reduce and supported tools</description>
```
Comment (gustavonalle) [inline core/pom.xml]: Commit should be called "Upgrade to Infinispan 9.4.1.Final", as "backport prod changes" does not mean anything :)

File: parent/pom.xml [hunk 2]
```diff
@@ -166,11 +183,87 @@
...
+                <artifactId>bom-builder-maven-plugin</artifactId>
+                <version>1.2.0</version>
+                <executions>
+                    <execution>
+                        <id>build-bom</id>
+                        <goals>
+                            <goal>build-bom</goal>
+                        </goals>
+                        <configuration>
+                            <bomGroupId>org.infinispan.gen.bom</bomGroupId>
```
Comment (gustavonalle) [inline parent/pom.xml]: What is this plugin used for and how to use it? I can see in that in the /target, the following file name bom-pom.xml is generated, which doesn't seem very useful:
```
<?xml version="1.0" encoding="UTF-8"?>
<project xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd" xmlns="http://maven.apache.org/POM/4.0.0"
    xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">
  <modelVersion>4.0.0</modelVersion>
  <groupId>org.infinispan.gen.bom</groupId>
  <artifactId>gen-bom</artifactId>
  <version>1.0.0</version>
  <packaging>pom</packaging>
  <name>Gen BOM</name>
  <description>Gen BOM</description>
  <properties>
    <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
  </properties>
  <dependencyManagement />
</project>
```

File: parent/pom.xml [hunk 3]
```diff
@@ -143,6 +153,13 @@
 
     <dependencyManagement>
         <dependencies>
+            <dependency>
+                <groupId>org.infinispan</groupId>
+                <artifactId>infinispan-bom</artifactId>
+                <version>${version.infinispan}</version>
```
Comment (gustavonalle) [inline parent/pom.xml]: This import is only used to inherit the version of ```org.infinispan:infinispan-client-hotrod``` and  ```org.infinispan:infinispan-core```, wouldn't be simpler to just use ```<version>${version.infinispan}</version>``` in those two dependencies?

**Review by ghost — COMMENTED** [review]
Looks good overall, I just have a couple of questions

File: parent/pom.xml (hunk 3 above)
Comment (pruivo) [reply parent/pom.xml]: meh. I'll change it... It was just to test if the bom is okish :)

File: parent/pom.xml (hunk 2 above)
Comment (jsenko) [reply parent/pom.xml]: Hi, the BOM plugin can be removed, i've used it to create the big infinispan BOM

**Comment by jsenko** [comment]
The parent was there to provide branding properties inherited from infinispan. If the parent is not included, the properties need to be explicitly modified.
I think the distribution profile is a good idea.

**Comment by jsenko** [comment]
I'll test this for DR4

**Comment by pruivo** [comment]
Updated with the following changes:
* removed pom generation
* javadoc runs on `core` when `distribution` profile is set (zip is created with the javadocs :+1: )
* re-added `infinispan-build-configuration-parent` and removed duplicated properties

**Comment by pruivo** [comment]
@gustavonalle should we use 9.4.1.Final ou SNAPSHOT?

**Comment by gustavonalle** [comment]
@pruivo SNAPSHOT should never be used for dependencies

**Comment by gustavonalle** [comment]
Merged, thanks @pruivo !

**Pull request merged**